code/game/objects/items/extinguisher.dm

```diff
@@ -143,64 +143,77 @@
 		var/direction = get_dir(src,target)
 
 		if(user.buckled && isobj(user.buckled) && !user.buckled.anchored)
-			spawn(0)
-				var/obj/B = user.buckled
-				var/movementdirection = turn(direction,180)
-				step(B, movementdirection)
-				sleep(1)
-				step(B, movementdirection)
-				sleep(1)
-				step(B, movementdirection)
-				sleep(1)
-				step(B, movementdirection)
-				sleep(2)
-				step(B, movementdirection)
-				sleep(2)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
-				sleep(3)
-				step(B, movementdirection)
+			var/obj/B = user.buckled
+			var/movementdirection = turn(direction,180)
+			addtimer(CALLBACK(src, /obj/item/extinguisher/proc/move_chair, B, movementdirection), 1)

```

Comment:
Timers can be 0 wait just like spawns

Reply:
The spawn(0) is only used for the loop on each water particle, as to make them move from tile to tile together instead of one after another. It's lazy coding, refusing to use another loop for making them move and using spawn(0).